src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -183,18 +184,27 @@ internal static class MethodGenerator
             return localFunctionDeclaration;
         }
 
-        private static SyntaxList<AttributeListSyntax> GenerateAttributes(
+        private static (SyntaxList<AttributeListSyntax>, NullableAnnotation) GenerateAttributes(
             IMethodSymbol method, CodeGenerationOptions options, bool isExplicit)
         {
-            var attributes = new List<AttributeListSyntax>();
+            var returnNullableAnnotation = method.ReturnNullableAnnotation;
 
-            if (!isExplicit)
+            if (isExplicit)
             {
-                attributes.AddRange(AttributeGenerator.GenerateAttributeLists(method.GetAttributes(), options));
-                attributes.AddRange(AttributeGenerator.GenerateAttributeLists(method.GetReturnTypeAttributes(), options, SyntaxFactory.Token(SyntaxKind.ReturnKeyword)));
+                return (default, returnNullableAnnotation);
```

Comment:
```suggestion
                // Add explanatory comment here.
                return (default, method.ReturnNullableAnnotation);
```

Reply:
TBH I don't know why explicit members do not care about attributes. Looks like they should.

Reply:
For example, if you explicitly implement `IEquatable<T>`, the `other` argument is definitely nullable.

#### Actual

```cs
bool IEquatable<string>.Equals(string other)
```

#### Expected

```cs
bool IEquatable<string>.Equals(string? other)
```